### PR TITLE
You can remove cuffed orange shoes with your hand

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Orange shoes and cuffs can now be removed.

## Why It's Good For The Game

These items are very meta and very powerful, given how easy they are to acquire.

There's no way to get rid of these shoes. They are great for keeping cuffed people unable to run off, but under no way should they be an absolute no escape item. Sec has plenty of ways to contain prisoners. This shouldn't be one of them.

## Changelog
:cl:
tweak: You can now remove your own shoecuffs.
/:cl:

### CODE REVIEW

File: code/modules/clothing/shoes/colour.dm [hunk 1]
```diff
@@ -98,27 +98,9 @@
 /obj/item/clothing/shoes/sneakers/orange/attackby(obj/H, loc, params)
 	..()
 	// Note: not using istype here because we want to ignore all subtypes
-	if (H.type == /obj/item/restraints/handcuffs && !chained)
+	if (!chained && istype(H,/obj/item/restraints/handcuffs))
```
Comment (PowerfulBacon) [inline code/modules/clothing/shoes/colour.dm]:
```suggestion
	if (!chained && H.type == /obj/item/restraints/handcuffs)
```

**Review by PowerfulBacon — COMMENTED** [review]
How about instead of making it so you can instantly remove them, have a do_after with a 60 second cooldown so you can remove them after standing still.

File: code/modules/clothing/shoes/colour.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/modules/clothing/shoes/colour.dm]: not all handcuffs make sense being used to make the cool shoes

**Comment by eeSPee** [comment]
> How about instead of making it so you can instantly remove them, have a do_after with a 60 second cooldown so you can remove them after standing still.

I don't know. I see them as a super slow that prevents cuffed people from running away, but they're not actual legcuffs; they slow you to a crawl, but are easy to remove.

**Comment by utilbird** [comment]
> How about instead of making it so you can instantly remove them, have a do_after with a 60 second cooldown so you can remove them after standing still.

This. although 60 seconds is a bit long imo

**Comment by eeSPee** [comment]
Like this?

File: code/modules/clothing/shoes/colour.dm [hunk 2]
```diff
@@ -98,27 +98,26 @@
...
-
+	
 /obj/item/clothing/shoes/sneakers/orange/allow_attack_hand_drop(mob/user)
 	if(ishuman(user))
 		var/mob/living/carbon/human/C = user
 		if(C.shoes == src && chained == 1)
-			to_chat(user, "<span class='warning'>You need help taking these off!</span>")
-			return FALSE
+			to_chat(user, "<span class='warning'>You start taking off your [src]!</span>")			
+			return do_after(user,15 SECONDS, src)
```
Comment (PowerfulBacon) [inline code/modules/clothing/shoes/colour.dm]:
```suggestion
			to_chat(user, "<span class='warning'>You start taking off your [src]!</span>")
			if(!do_after(user,15 SECONDS, src))
				return FALSE
```
Something like this should work

**Review by PowerfulBacon — DISMISSED** [review]

File: code/modules/clothing/shoes/colour.dm [hunk 3]
```diff
@@ -98,27 +98,28 @@
...
-
+	
 /obj/item/clothing/shoes/sneakers/orange/allow_attack_hand_drop(mob/user)
 	if(ishuman(user))
 		var/mob/living/carbon/human/C = user
 		if(C.shoes == src && chained == 1)
-			to_chat(user, "<span class='warning'>You need help taking these off!</span>")
-			return FALSE
+			to_chat(user, "<span class='warning'>You start taking off your [src]!</span>")
+			if(!do_after(c,15 SECONDS, src))
```
Comment (PowerfulBacon) [inline code/modules/clothing/shoes/colour.dm]:
```suggestion
			if(!do_after(C,15 SECONDS, src))
```

File: code/modules/clothing/shoes/colour.dm [hunk 4]
```diff
@@ -98,27 +98,28 @@
...
 
 /obj/item/clothing/shoes/sneakers/orange/MouseDrop(atom/over)
 	var/mob/m = usr
 	if(ishuman(m))
 		var/mob/living/carbon/human/c = m
 		if(c.shoes == src && chained == 1)
-			to_chat(c, "<span class='warning'>You need help taking these off!</span>")
-			return
+			to_chat(user, "<span class='warning'>You start taking off your [src]!</span>")			
+			if(!do_after(c,15 SECONDS, src))
```
Comment (PowerfulBacon) [inline code/modules/clothing/shoes/colour.dm]:
```suggestion
			if(!do_after(C,15 SECONDS, src))
```

File: code/modules/clothing/shoes/colour.dm [hunk 5]
```diff
@@ -98,27 +98,28 @@
...
 
 /obj/item/clothing/shoes/sneakers/orange/MouseDrop(atom/over)
 	var/mob/m = usr
 	if(ishuman(m))
 		var/mob/living/carbon/human/c = m
 		if(c.shoes == src && chained == 1)
-			to_chat(c, "<span class='warning'>You need help taking these off!</span>")
-			return
+			to_chat(user, "<span class='warning'>You start taking off your [src]!</span>")			
+			if(!do_after(C,15 SECONDS, src))
```
Comment (PowerfulBacon) [inline code/modules/clothing/shoes/colour.dm]:
```suggestion
			if(!do_after(c,15 SECONDS, src))
```
this one is a lowercase but the top one is uppercase. wow. Whoever coded this before was indeed a true coder.

**Comment by FortSumter** [comment]
Nerfing Security bad. Reject Disablers, return to Glock.

File: code/modules/clothing/shoes/colour.dm [hunk 6]
```diff
@@ -98,27 +98,28 @@
...
-
+	
 /obj/item/clothing/shoes/sneakers/orange/allow_attack_hand_drop(mob/user)
 	if(ishuman(user))
-		var/mob/living/carbon/human/C = user
-		if(C.shoes == src && chained == 1)
-			to_chat(user, "<span class='warning'>You need help taking these off!</span>")
-			return FALSE
+		var/mob/living/carbon/human/hummie = user
+		if(hummie.shoes == src && chained == 1)
```
Comment (PowerfulBacon) [inline code/modules/clothing/shoes/colour.dm]:
```suggestion
		if(hummie.shoes == src && chained)
```

File: code/modules/clothing/shoes/colour.dm [hunk 7]
```diff
@@ -98,27 +98,28 @@
...
 
 /obj/item/clothing/shoes/sneakers/orange/MouseDrop(atom/over)
 	var/mob/m = usr
 	if(ishuman(m))
-		var/mob/living/carbon/human/c = m
-		if(c.shoes == src && chained == 1)
-			to_chat(c, "<span class='warning'>You need help taking these off!</span>")
-			return
+		var/mob/living/carbon/human/hummie = m
+		if(hummie.shoes == src && chained == 1)
```
Comment (PowerfulBacon) [inline code/modules/clothing/shoes/colour.dm]:
```suggestion
		if(hummie.shoes == src && chained)
```

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**